src/tests/test_dynamic_api/test_views_api.py

```diff
@@ -1804,67 +1805,147 @@ def test_unpaginated_list(self, format, api_client, afval_container, filled_rout
+                "href": f"http://testserver/v1/afvalwegingen/containers"
+                f"/?_format=json&{page_size_param}={self.PAGE_SIZE}"
+            }
+
+        return page
+
+    def _make_geojson_page(self, page_num: int, page_size_param: str):
+        start = 1 + (page_num - 1) * self.PAGE_SIZE
+        end = start + self.PAGE_SIZE
+
+        page = {
+            "type": "FeatureCollection",
+            "crs": {
+                "properties": {"name": "urn:ogc:def:crs:EPSG::4326"},
+                "type": "name",
+            },
+            "features": [
+                {
+                    "type": "Feature",
+                    "id": f"containers.{i}",
+                    "geometry": {
+                        "type": "Point",
+                        "coordinates": GEOJSON_POINT,
+                    },
+                    "properties": {
+                        "id": i,
+                        "clusterId": "c1",
+                        "serienummer": "foobar-123",
+                        "datumCreatie": "2021-01-03",
+                        "eigenaarNaam": "Dataservices",
+                        "datumLeegmaken": "2021-01-03T12:13:14",
+                    },
+                }
+                for i in range(start, end)
+            ],
+            "_links": [
+                {
+                    "rel": "next",
+                    "type": "application/geo+json",
+                    "title": "next page",
+                    "href": "http://testserver/v1/afvalwegingen/containers"
+                    f"/?_format=geojson&_pageSize=4&page={page_num+1}",
+                },
+            ],
+        }
+
+        if page_num > 1:
+            page["_links"].append(
+                {
+                    "rel": "previous",
+                    "type": "application/geo+json",
+                    "title": "previous page",
+                    "href": "http://testserver/v1/afvalwegingen/containers"
+                    "/?_format=geojson&_pageSize=4",
+                },
+            )
+
+        return page
+
+    PAGINATED_FORMATS = {
+        # "csv": (b''.join, "text/csv; charset=utf-8", _make_csv_page),
+        "csv": (as_is, "text/csv; charset=utf-8", _make_csv_page),
+        "json": (orjson.loads, "application/hal+json", _make_json_page),
+        "geojson": (orjson.loads, "application/geo+json; charset=utf-8", _make_geojson_page),
     }
 
     @pytest.mark.parametrize("format", sorted(PAGINATED_FORMATS.keys()))
+    @pytest.mark.parametrize("page_num", (1, 2))
     @pytest.mark.parametrize("page_size_param", ["_pageSize", "page_size"])
     def test_paginated_list(
-        self, format, page_size_param, api_client, afval_container, filled_router
+        self, format, page_num, page_size_param, api_client, afval_container, filled_router
     ):
         """Prove that the pagination still works if explicitly requested."""
         decoder, expected_type, expected_data = self.PAGINATED_FORMATS[format]
```

Comment:
Maybe give `expected_data` a better name, so that it is clearer that it is a callable.

Reply:
Renamed to `make_expected`, please re-review.